From bf40cb5d11d65488dba0eebb8918ce3f72d4b094 Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Mon, 4 May 2026 19:28:50 +0800
Subject: [PATCH] dts: arm: ast2705: add XGMAC node

Add the AST2705 XGMAC node with the SOC1 clock, MAC and XPCS
resets, queue setup, and XPCS MDIO child. Point ethernet2 at the
XGMAC instance so boards can enable the 10G MAC through the alias.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: Id48498fe8f565bbd1c8b10f532f3d632993403b7
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 arch/arm/dts/ast2705.dtsi | 63 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 63 insertions(+)

diff --git a/arch/arm/dts/ast2705.dtsi b/arch/arm/dts/ast2705.dtsi
index d5aeac423e..ee3174d627 100644
--- a/arch/arm/dts/ast2705.dtsi
+++ b/arch/arm/dts/ast2705.dtsi
@@ -2,7 +2,70 @@
 #include "ast2700.dtsi"
 
 / {
+	aliases {
+		ethernet2 = &xgmac;
+	};
+
 	soc1: soc@14000000 {
+		xgmac: ethernet@140a0000 {
+			compatible = "aspeed,ast2700-dwxgmac", "intel,socfpga-dwxgmac",
+				     "snps,dwxgmac-2.10", "snps,dwxgmac";
+			reg = <0x0 0x140a0000 0x0 0x8000>;
+			interrupts = <GIC_SPI 2 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "macirq";
+			phy-mode = "10gbase-r";
+			rx-fifo-depth = <4096>;
+			tx-fifo-depth = <4096>;
+			clocks = <&soc1_clk SCU1_CLK_GATE_MAC2CLK>;
+			clock-names = "stmmaceth";
+			resets = <&soc1_rst SCU1_RESET_MAC2>,
+				 <&soc1_rst SCU1_RESET_XPCS>;
+			reset-names = "stmmaceth", "ahb";
+			snps,axi-config = <&xgmac_axi_setup>;
+			snps,mtl-rx-config = <&xgmac_mtl_rx_setup>;
+			snps,mtl-tx-config = <&xgmac_mtl_tx_setup>;
+			snps,tso;
+			snps,pbl = <8>;
+			phy-handle = <&xpcs0>;
+			pcs-handle = <&xpcs0>;
+			status = "disabled";
+
+			xgmac_axi_setup: stmmac-axi-config {
+				snps,wr_osr_lmt = <0xf>;
+				snps,rd_osr_lmt = <0xf>;
+				snps,kbbe;
+				snps,blen = <0 0 0 0 0 8 4>;
+			};
+
+			xgmac_mtl_rx_setup: rx-queues-config {
+				snps,rx-queues-to-use = <1>;
+				snps,rx-sched-sp;
+
+				queue0 {
+					snps,dcb-algorithm;
+					snps,map-to-dma-channel = <0>;
+				};
+			};
+
+			xgmac_mtl_tx_setup: tx-queues-config {
+				snps,tx-queues-to-use = <1>;
+
+				queue0 {
+					snps,dcb-algorithm;
+				};
+			};
+
+			xgmac_mdio0: mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "snps,dwmac-mdio";
+
+				xpcs0: ethernet-phy@0 {
+					reg = <0>;
+				};
+			};
+		};
+
 		syscon1: syscon@14c02000 {
 			compatible = "aspeed,ast2705-scu1", "syscon", "simple-mfd";
 			reg = <0x0 0x14c02000 0x0 0x1000>;
-- 
2.43.0

